lhotse/recipes/radio.py

```diff
@@ -0,0 +1,141 @@
+"""
+This recipe prepares data collected from radio streamed on the web. The data
+have some metadata attached to them, including the geographic location of
+broadcast, date and time of the recorded clip, as well as a unique station
+identifier.
+
+Obtaining the data
+-----------------------------------------------------------
+If you want to use this corpus please email: wiesner@jhu.edu
+
+As the data are collected from radio stream, they cannot be broadly
+disseminated or used for commercial purposes. In the email, include your
+affiliated academic institution and the intended use for the data and we will
+the data to you if it is indeed for non-commercial, academic purporses.
+
+Description
+------------------------------------------------------------
+The data consist of ∼4000 hours of speech collected between
+September 27, 2023 to October 1, 2023, in 9449 locations all over the world,
+from 17171 stations. 
+
+These data were used for Geolocation of speech in order to answer the question,
+Where are you from? in the paper 
+
+Where are you from? Geolocating Speech and Applications to Language
+Identification, presented at NAACL 2024. Please read for a full descrption
+and please cite as 
+ 
+@inproceedings{foley2024you,
+  title={Where are you from? Geolocating Speech and Applications to Language Identification},
+  author={Foley, Patrick and Wiesner, Matthew and Odoom, Bismarck and Perera, Leibny Paola Garcia and Murray, Kenton and Koehn, Philipp},
+  booktitle={Proceedings of the 2024 Conference of the North American Chapter of the Association for Computational Linguistics: Human Language Technologies (Volume 1: Long Papers)},
+  pages={5114--5126},
+  year={2024}
+}
+"""
+import json
+import re
+from functools import partial
+from pathlib import Path
+from typing import Dict, Optional, Union
+
+from tqdm import tqdm
+
+from lhotse.audio import Recording, RecordingSet, set_ffmpeg_torchaudio_info_enabled
+from lhotse.parallel import parallel_map
+from lhotse.supervision import SupervisionSegment, SupervisionSet
+from lhotse.utils import Pathlike
+
+set_ffmpeg_torchaudio_info_enabled(False)
```

Comment:
One last request - don’t set these things in the global scope because it’ll be executed when Lhotse is imported. Move it to local scope (even better i think this one works like a context manager)

Reply:
I think I fixed it now?

Reply:
Almost - now you need to move to actual usage to inside the function that does the audio reading and use it with Python with statement

Reply:
I actually forget why I started adding this in to all the Lhotse recipes. I think it fixed some problem I was having at some point, but I have no recollection of what that problem was. Do I actually need it? When I actually use it locally each time it gives me a ton of horrible logging messages.

Reply:
You shouldn’t need it. We are using libsndfile now by default rather than torchaudio. Feel free to remove it if you prefer

Reply:
That is what I ended up doing. I believe the current updated commit just removed this.